tools/advisory-parser/src/models.rs

```diff
@@ -0,0 +1,182 @@
+//! Advisory data models and deserialization logic.
+//! This module defines the structure of security advisory TOML files and implements
+//! custom deserialization with validation for CVE and GHSA identifiers.
+
+use regex::Regex;
+use serde::de::{self, MapAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+use std::fmt;
+
+#[derive(Deserialize)]
+pub struct Advisory {
+    pub advisory: AdvisoryInfo,
+}
+
+pub struct AdvisoryInfo {
+    #[allow(dead_code)]
+    /// Unique identifier for the Bottlerocket Security Advisory
+    pub id: String,
+
+    #[allow(dead_code)]
+    /// Title describing the security issue
+    pub title: String,
+
+    #[allow(dead_code)]
+    /// CVE identifier (e.g., "CVE-2024-1234")
+    pub cve: Option<String>,
+
+    #[allow(dead_code)]
+    /// GitHub Security Advisory identifier (e.g., "GHSA-xxxx-xxxx-xxxx")
+    pub ghsa: Option<String>,
+
+    #[allow(dead_code)]
+    /// Severity level (e.g., "high", "medium", "low")
```

Comment:
nit: Medium works, but usually we do: 
```suggestion
    /// Severity level (e.g., "high", "moderate", "low")
```